go.mod

```
@@ -1,6 +1,6 @@
 module github.com/grpc-ecosystem/grpc-gateway/v2
 
-go 1.24
+go 1.23.0
```

If the suffix ".0" wasn't automatically added by `go mod tidy`, then we should prefer to `1.23`. You can also add `toolchain go1.23.6` if you want.
```suggestion
go 1.23
```

---

This because the go version of package `google.golang.org/genproto/googleapis/rpc` is declared as `1.23.0`, so `1.23.0` should be the minimum version available here.

---

OK, then it's automatically added by `go mod tidy`. Technically, I don't think there is any difference between `go 1.23` and `go 1.23.0`; but Semantically, `go 1.23.x` means it requires the minimum go patch version for the downstream modules.

I don't know what motivate some people to add the suffix ".0". On the other hand, the golang team should treat the `go 1.x.0` equivalent to `go 1.x`, and do not require the downstream module to add the suffix no matter the upstream has it or not.

---

[Starting in Go 1.21](https://go.dev/doc/go1.21#introduction), the go directive in the go.mod file supports specifying minor versions, enhancing versioning precision while maintaining backward compatibility.
Setting it to `1.23.0` means that downstream modules must use a Go higher than `1.23.0`(e.g. `1.23.6`, `1.24.0`), not just `1.23.0`.